script.js

```diff
@@ -1,12 +1,543 @@
+
+  // access the data through the array
+  episodes.forEach((data) => {
+    if (
+      data.name.toLowerCase().includes(searchString) ||
+      data.summary.toLowerCase().includes(searchString)
+    ) {
+      episodeCount++;
+      showEpisode(episodeContainers[episodeIndex]);
+    } else if (
+      data.name == null ||
+      data.name == "" ||
+      data.summary == null ||
+      data.summary == ""
+    ) {
+      hideEpisode(episodeContainers[episodeIndex]);
+    } else {
+      hideEpisode(episodeContainers[episodeIndex]);
+    }
+    episodeIndex++;
+  });
+  updateCounter(episodeCount, episodes.length);
+}
+
+const addSearchBox = (episodeList) => {
+  let inputBox = document.getElementById("inputBoxId");
+  if (inputBox === null) {
+    // this only works if the ""inputBox does not exist
+    inputBox = document.createElement("input");
+    inputBox.setAttribute("id", "inputBoxId");
+    inputBox.setAttribute("placeHolder", "type and search");
+    inputBox.style.width = "25em";
+    inputBox.style.backgroundColor = "#f5f5f5";
+    inputBox.style.margin = "0.8em"; // spacing to the left of episode list
+  }
+
+  inputBox.addEventListener("keyup", (event) => {
+    const searchString = event.target.value.toLowerCase();
+    let tvSelectDom = document.querySelector("#select-tv");
+
+    if (tvSelectDom.style.display === "none") {
+      // this checks whether the tv show or the episodes state exist, if so we call the data function
+      searchEpisodeData(searchString); // every time you type data the inbox addEventListener is called
+    } else {
+      searchTvShowData(searchString);
+    }
+  });
+
+  return inputBox;
+};
+
+const addHeader = (showData) => {
+  // changed the order of arguments // removed episodeList to ensure general purpose use
+  const rootElem = document.getElementById("root");
+  let body = document.querySelector("body");
+
+  let navBarContainer = document.getElementById("navBarContainer");
+
+  if (navBarContainer === null) {
+    //
+    navBarContainer = document.createElement("div");
+    navBarContainer.style.display = "flex";
+    navBarContainer.setAttribute("id", "navBarContainer");
+
+    body.insertBefore(navBarContainer, rootElem);
+
+    const showSelector = addShowSelector(showData);
+    navBarContainer.appendChild(showSelector);
+
+    const inputBox = addSearchBox(showData); // changed it from episodeList
+    navBarContainer.appendChild(inputBox);
+    const counter = addCounter();
+
+    navBarContainer.appendChild(counter);
+    updateCounter(showData.length, showData.length); // how many are we showing/ how many are there in total
+  }
+
+  const dropDown = addDropDownMenu();
+  navBarContainer.appendChild(dropDown);
+};
+
+function makePageForEpisodes(episodeList) {
+  episodes = episodeList;
+  const tvShowData = getAllShows();
+  addAllEpisodes(episodeList);
+  addHeader(episodeList, tvShowData);
+  addFooter();
+}
+
+function setup() {
```

Comment:
This function is never used, so it can be deleted.

Reply:
I have removed it now thanks for the feedback.